azure-iot-device/azure/iot/device/iothub/sync_clients.py

```diff
@@ -323,6 +323,47 @@ def receive_twin_desired_properties_patch(self, block=True, timeout=None):
         logger.info("twin patch received")
         return patch
 
+    def get_blob_shared_access_signature(self, blob_name):
```

Comment:
>get_blob_shared_access_signature [](start = 8, length = 32)

Is this function only returning a sas?  Or is it returning more?  If this returns an access signature, what else does the caller need to actually upload.  It looks like there's a bunch of logic in the sample apps that could maybe be put into the client.

Reply:
this returns a sas and a correlation_id in an object. I've updated the api name to get_storage_info, since I think that's a better option.

---
In reply to: [352787099](https://github.com/Azure/azure-iot-sdk-python/pull/403#discussion_r352787099) [](ancestors = 352787099)